src/Arcus.EventGrid.Tests.Integration/Publishing/EventPublishingTests.cs

```diff
@@ -191,6 +220,78 @@ await EventGridPublisherBuilder
                 });
         }
 
+        [Fact]
+        public async Task PublishMultipleRawEvents_WithBuilder_ValidParameters_SucceedsWithRetryCount()
+        {
+            // Arrange
+            var topicEndpoint = Configuration.GetValue<string>("Arcus:EventGrid:TopicEndpoint");
+            var endpointKey = Configuration.GetValue<string>("Arcus:EventGrid:EndpointKey");
+            var events =
+                Enumerable
+                    .Repeat<Func<Guid>>(Guid.NewGuid, 2)
+                    .Select(newGuid => new RawEvent(
+                                newGuid().ToString(),
+                                subject: "integration-test",
+                                body: "{\"licensePlate\": \"1-TOM-1337\"}",
+                                type: "Arcus.Samples.Cars.NewCarRegistered",
+                                dataVersion:"1.0",
+                                eventTime: DateTimeOffset.Now))
+                    .ToArray();
+
+            // Act
+            await EventGridPublisherBuilder
+                  .ForTopic(topicEndpoint)
+                  .UsingAuthenticationKey(endpointKey)
+                  .Build()
+                  .PublishManyAsync(events);
+
+            // Assert
+            Assert.All(
+                events,
+                rawEvent =>
+                {
+                    TracePublishedEvent(rawEvent.Id, events);
+                    string receivedEvent = _serviceBusEventConsumerHost.GetReceivedEvent(rawEvent.Id, 5);
+                    AssertReceivedEvent(rawEvent.Id, rawEvent.EventType, rawEvent.Subject, JToken.FromObject(rawEvent.Data)["licensePlate"].ToString(), receivedEvent);
+                });
+        }
+
+        [Fact]
+        public async Task PublishMultipleRawEvents_WithBuilder_ValidParameters_SucceedsWithTimeout()
+        {
+            // Arrange
+            var topicEndpoint = Configuration.GetValue<string>("Arcus:EventGrid:TopicEndpoint");
+            var endpointKey = Configuration.GetValue<string>("Arcus:EventGrid:EndpointKey");
+            var events =
+                Enumerable
+                    .Repeat<Func<Guid>>(Guid.NewGuid, 2)
+                    .Select(newGuid => new RawEvent(
+                                newGuid().ToString(),
+                                subject: "integration-test",
+                                body: "{\"licensePlate\": \"1-TOM-1337\"}",
+                                type: "Arcus.Samples.Cars.NewCarRegistered",
+                                dataVersion:"1.0",
+                                eventTime: DateTimeOffset.Now))
+                    .ToArray();
+
+            // Act
+            await EventGridPublisherBuilder
+                  .ForTopic(topicEndpoint)
+                  .UsingAuthenticationKey(endpointKey)
+                  .Build()
+                  .PublishManyRawAsync(events);
+
+            // Assert
+            Assert.All(
```

Comment:
Would prefer to use the traditional approach or at least move all the assertion for a raw event to a method rather than doing everything inline